src/timesync/timesyncd-server.c

```diff
@@ -54,6 +54,7 @@ int server_name_new(
                 const char *string) {
 
         ServerName *n, *tail;
+        const char *type_name = NULL;
```

Comment:
consider using DEFINE_STRING_TABLE_LOOKUP_TO_STRING() to define lookup function with table, and use that.